### PR TITLE
refactor(helper): replace mutateMe usage [PART-1]

### PR DESCRIPTION
**Pull request by samouss**
The PR replaces the usage of the `mutateMe` function. This function has been removed from the latest version of the Helper. We leverage the internals of the Helper to remove the parameters added by the widget. You can verify that the tests pass for the connector with:

```
yarn jest src/connectors/configure/__tests__/connectConfigure-test.js
```

> The CI does not pass, it's expected. It won't pass until the last PR of the stack.

### CODE REVIEW

File: src/connectors/configure/connectConfigure.js [hunk 1]
```diff
@@ -91,13 +91,18 @@ export default function connectConfigure(renderFn = noop, unmountFn = noop) {
...
       removeSearchParameters(state) {
-        // widgetParams are assumed 'controlled',
-        // so they override whatever other widgets give the state
-        return state.mutateMe(mutableState => {
-          Object.keys(widgetParams.searchParameters).forEach(key => {
-            delete mutableState[key];
-          });
-        });
+        // We leverage the Helper internals to remove the `widgetParams` from
+        // the state. The function `setQueryParameters` omit (or remove) the
```
Comment (Haroenv) [inline src/connectors/configure/connectConfigure.js]:
```suggestion
        // the state. The function `setQueryParameters` omits the
```

This wording is slightly clearer to me

Comment (samouss) [reply src/connectors/configure/connectConfigure.js]: https://github.com/algolia/instantsearch.js/pull/3792/commits/bc7c101f0b228cf2a6bb92e2485679567a238189

**Comment by Haroenv** [comment]
looks like the history here is wrong now :)

**Comment by samouss** [comment]
@Haroenv It should be good now, looks like GitHub wasn't able to detect that the base branch already contained the previous commits.

File: src/connectors/configure/connectConfigure.js [hunk 2]
```diff
@@ -91,13 +91,18 @@ export default function connectConfigure(renderFn = noop, unmountFn = noop) {
...
-        // widgetParams are assumed 'controlled',
-        // so they override whatever other widgets give the state
-        return state.mutateMe(mutableState => {
-          Object.keys(widgetParams.searchParameters).forEach(key => {
-            delete mutableState[key];
-          });
-        });
+        // We leverage the Helper internals to remove the `widgetParams` from
+        // the state. The function `setQueryParameters` omits the values that
+        // are `undefined` on the next state.
```
Comment (tkrugg) [inline src/connectors/configure/connectConfigure.js]: I'm not sure I understand what you mean.
Is it that for example

```js
const state = { a: 'something', b: 'someotherthing' }
const nextState = state.setQueryParameters({ a: undefined })
// nextState is { b: 'someotherthing' }
```

Comment (tkrugg) [reply src/connectors/configure/connectConfigure.js]: I tried writing a test for it and it fails.
https://github.com/algolia/algoliasearch-helper-js/compare/drafts/setQueryParameters

Comment (Haroenv) [reply src/connectors/configure/connectConfigure.js]: you need to compare to the `next` branch, it already passes & has a test there: 

https://github.com/algolia/algoliasearch-helper-js/blob/ffa13f45dd4a71d0a148ecd72fa1da6fb8da4c4f/test/spec/SearchParameters/setQueryParameters.js#L77-L91

https://github.com/algolia/algoliasearch-helper-js/compare/next...drafts/setQueryParameters

**Pull request merged**